django_irods/views.py

```diff
@@ -100,16 +98,14 @@ def download(request, path, rest_call=False, use_async=True, use_reverse_proxy=T
             response.content = "<h1>" + content_msg + "</h1>"
             return response
 
-    # default values are changed later as needed
-
     istorage = res.get_irods_storage()
-    if res.is_federated:
-        irods_path = os.path.join(res.resource_federation_path, path)
-    else:
-        irods_path = path
+    # res.resource_federation_path would only be non-empty when potentially third-party federated
+    # zones join HydroShare. The line below would work currently when res.resource_federation_path
+    # is empty and in the future when a third-party federated zone potentially joins HydroShare
+    irods_path = os.path.join(res.resource_federation_path, path)
```

Comment:
Is this resource method has any use here?
https://github.com/hydroshare/hydroshare/blob/develop/hs_core/models.py#L1960

Reply:
This one took me a bit to understand, but the comment explains it pretty clearly.  If the resource isn't federated, it'll resovle to `irods_path = path` because `res.resource_federation_path` will be empty.  I guess we could be asking ourselves if this is easier to read than before?

I'd like to wrap all concepts of federation within the Resource and ResourceFile classes if possible.  This change just drops the number of lines and leaves federation concepts in this view.

Reply:
@pkdash @sblack-usu thanks for pointing out that resource method ```get_irods_path``` which I believe could be leveraged here with some customization. That method assumes short_id is not included in the path, but in this case, short_id indeed is included in the path, but I can do a check in that method to accommodate both uses. By the way, do you think ```get_public_path()``` call in this line: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/views/resource_folder_hierarchy.py#L85 should be replaced by ```get_irods_path``` as well? If that can be replaced, we could either remove this ```get_public_path()``` or make it an internal method since it is pretty confusing to me to have all these public get path related methods on resource object.

Reply:
@hyi If the ```get_public_path()``` is used only at that one place, then probably we can replace that with the ```get_irods_path()``` and remove the ```get_public_path()``` from the resource class. May be @alvacouch have some thoughts on this as he probably implemented these methods in resource class.

Reply:
@pkdash Yes, that is the only place this ```get_public_path()``` is called externally in addition to being called by ```get_irods_path()``` as an internal method. @alvacouch I will consolidate these two into one method ```get_irods_path()``` unless you disagree.